server/src/main/java/org/elasticsearch/transport/OutboundHandler.java

```diff
@@ -195,6 +195,7 @@ public BytesReference get() throws IOException {
                 TransportLogger.logOutboundMessage(channel, message);
                 return message;
             } catch (Exception e) {
+                assert false : new AssertionError("Serialization shouild never fail", e);
```

Comment:
I'm not convinced about this. There are some places where we throw an exception because we're trying to serialise something to a version of Elasticsearch that simply cannot understand the message we want to send. For instance, this:

https://github.com/elastic/elasticsearch/blob/9acd2fd1fd1130d6c8a6bfc6e19eb9af7a44621a/server/src/main/java/org/elasticsearch/common/lucene/Lucene.java#L417

I think an exception is a reasonable way to handle this kind of impossible-to-represent message.

Reply:
Isn't the spot you're linking a bug and a situation we should never get into? (why would an old node request top hits that it can't understand?) Why would we try to send a node a message that it can't understand? I know we had similar logic in snapshots with the multi-snapshot get and it was always a bug to get to the point of throwing there.

Reply:
Eh possibly in that case, what about things like this?

https://github.com/elastic/elasticsearch/blob/3a64ecb6bfe5c7d88ac73954a3b498cc71ab8d6e/server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/delete/DeleteSnapshotRequest.java#L88-L89

We'd hit that if we were trying to delete multiple snapshots in a cluster with the coordinating node on 7.8 and the master on 7.7 IIUC. Rejecting the request seems reasonable, you can't do this until the upgrade is complete, but rejecting it elsewhere would be pretty painful.

Reply:
Fair point, forgot all about that direction of BwC => I dropped the assert